### ISSUE TITLE
🐛 noUselessElse rule incorrectly detect `else if` statements

### ISSUE THREAD

**Issue by EvHaus** [biomejs/biome#558]
### Environment information

```block
Incompatible Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

ℹ Rage discovered this running server using an incompatible version of Biome.

Server:
  Version:                      1.2.2
```

### What happened?

When the `nursery/noUselessElse` rule is enabled, it incorrectly detects `else if` statements as useless. For example:

```js
const formatter = () => {
	if (country === 'CANADA') {
		return 'Canada';
	// biome flags this as "This else clause can be omitted"
	} else if (country === 'USA') {
		return 'USA';
	}
}
```

### Expected result

An `else if` statement, like the one in my example, is not safe to be ommitted without changing the logic of the function. I think the rule should ignore `else if` statements.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
The rule proposes the following refactoring:

```js
const formatter = () => {
	if (country === 'CANADA') {
		return 'Canada';
	}
        if (country === 'USA') {
		return 'USA';
	}
}
```

This has the same behavior as before?

**Comment by EvHaus** [comment]
My bad, that makes perfect sense after looking at it more closely. Thanks @Conaclos.
